esmvaltool/utils/recipe_filler.py

```diff
@@ -0,0 +1,684 @@
+
+
+def get_args():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('recipe', help='Path/name of yaml pilot recipe file')
+    parser.add_argument('-c',
+                        '--config-file',
+                        default=os.path.join(os.environ["HOME"], '.esmvaltool',
+                                             'config-user.yml'),
+                        help='User configuration file')
+
+    parser.add_argument('-o',
+                        '--output',
+                        default=os.path.join(os.getcwd(),
+                                             'recipe_autofilled.yml'),
+                        help='Output recipe, default recipe_autofilled.yml')
+
+    args = parser.parse_args()
+    return args
+
+
+def _get_timefiltered_files(recipe_dict, exps_list, cmip_era):
+    """Obtain all files that correspond to requested time range."""
+    # multiple experiments allowed, complement data from each exp
+    if len(exps_list) > 1:
+        files = []
+        for exp in exps_list:
+            recipe_dict["exp"] = exp
+            files.extend(list_all_files(recipe_dict, cmip_era))
+        files = filter_years(files,
+                             recipe_dict["start_year"],
+                             recipe_dict["end_year"],
+                             overlap=True)
+        recipe_dict["exp"] = exps_list
+
+    else:
+        files = list_all_files(recipe_dict, cmip_era)
+        files = filter_years(files, recipe_dict["start_year"],
+                             recipe_dict["end_year"])
+
+    return files
+
+
+def run():
+    """Run the `recipe_filler` tool. Help in __doc__ and via --help."""
+    # Get arguments
+    args = get_args()
+    input_recipe = args.recipe
+    output_recipe = args.output
+    cmip_eras = ["CMIP5", "CMIP6"]
+
+    # read the config file
+    config_user = read_config_user_file(args.config_file,
+                                        'recipe_filler',
+                                        options={})
+
+    # configure logger
+    run_dir = os.path.join(config_user['output_dir'], 'recipe_filler')
+    if not os.path.isdir(run_dir):
+        os.makedirs(run_dir)
+    log_files = configure_logging(output_dir=run_dir,
+                                  console_log_level=config_user['log_level'])
+    logger.info(HEADER)
+    logger.info(f"Using user configuration file: {args.config_file}")
+    logger.info(f"Using pilot recipe file: {input_recipe}")
+    logger.info(f"Writing filled out recipe to: {output_recipe}")
+    log_files = "\n".join(log_files)
+    logger.info(f"Writing program log files to:\n{log_files}")
+
+    # check config user file
+    _check_config_file(config_user)
+
+    # parse recipe
+    with open(input_recipe, 'r') as yamlfile:
+        yamlrecipe = yaml.safe_load(yamlfile)
+        _check_recipe(yamlrecipe)
```

Comment:
We have already have a recipe checker in `esmvalcore`: https://github.com/ESMValGroup/ESMValCore/blob/master/esmvalcore/_recipe_checks.py

Can we use `check.recipe_with_schema(yamlfile)` to check the recipe, and implement needed checks there?

Reply:
again, not quite the same as that, it's a bit less stringent this one